Remove trailing whitespace for python and c++ generated code (Fix 112)

KrystianKedron opened the pull request:
#112 

This commit changes a little bit the PR for c++, in most cases, we will have:

```cpp
/*----- PROTECTED REGION ID(Test::write_spectrumDynamicTest) ENABLED START -----*/
//	Add your own code
/*----- PROTECTED REGION END -----*/	//	Test::write_spectrumDynamicTest
```

Or:

```cpp
/*----- PROTECTED REGION ID(Test::write_spectrumDynamicTest) ENABLED START -----*/
/*----- PROTECTED REGION END -----*/	//	Test::write_spectrumDynamicTest
```

To get rid of trailing whitespace from PR blocks.

KrystianKedron committed:
Remove trailing whitespace in python HL

KrystianKedron committed:
Remove trailing white-space in license section

 For python generated code

KrystianKedron committed:
Remove trailing white-space in PR section

KrystianKedron committed:
Remove trailing white-space for python generated code

- in __init__ section
- in read/write section
- in initialize_dynamic_attributes section
- in dyn_attr section
- in main() section

KrystianKedron committed:
Fix typo in the main() method

KrystianKedron committed:
Remove trailing white-space in c++ headers

KrystianKedron committed:
Remove trailing white-space in DeviceClassInclude

KrystianKedron committed:
Remove trailing white-space in c++ DeviceSource class

KrystianKedron committed:
Remove trailing white-space in c++ DynamicAtttributeUtils

KrystianKedron committed:
Remove trailing white-space in c++ ProtectedArea

KrystianKedron committed:
Various fixes for sonar and typos

dlacoste-esrf reviewed (commented):
Thanks a lot, there is a lot of typos fixed as well, this is really nice !

it raises some questions on the indentations conventions and such though.

KrystianKedron committed:
Address review commits

- Fix indication from PythonDevice.xtend DeviceClassInclude.xtend
- Improve protectedArea for Pipes.xtend
